core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,659 @@
+    submitted_on = datastore_services.DateTimeProperty(
+        required=True, indexed=True)
+    # The type of feedback for this report; choices are not specified as
+    # iterations of the report structure may introduce new types and we cannot
+    # rely on the backend updates to fully sync with the frontend report
+    # updates.
+    report_type = datastore_services.StringProperty(required=True, indexed=True)
+    # The category that this feedback is for.
+    category = datastore_services.TextProperty(required=True, indexed=False)
+    # The version of the app; on Android this is the package version name (e.g.
+    # 1.0-release-arm...) and on web this is the release version (e.g. 3.0.8).
+    platform_version = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The user's country locale represented as a ISO-3166 code; the locale is
+    # determined by the user's Android device settings.
+    device_country_locale_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The entry point location that the user is accessing the feedback report
+    # from on both web & Android devices. On Android, this could be
+    # navigation_drawer, lesson_player, revision_card, or crash.
+    entry_point = datastore_services.StringProperty(required=True, indexed=True)
+    # Additional topic / story / exploration IDs that may be collected depending
+    # on  the entry_point used to send the report; lesson player will have
+    # topic_id, story_id, and exploration_id, while revision card while have
+    # topic_id and subtopic_id.
+    entry_point_topic_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_story_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_exploration_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_subtopic_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+
+    # The text language on Oppia set by the user in its ISO-639 language code;
+    # this is set by the user in Oppia's app preferences. Current languages on
+    # Android can be: EN, FR, HI, or ZH.
+    text_language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The audio language ISO-639 code on Oppia set by the user as declared in
+    # the app. Current supported languages: English, Hindi, and Hinglish.
+    audio_language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+
+    # The Android device model used to submit the report.
+    android_device_model = datastore_services.StringProperty(
+        required=False, indexed=True)
+    # The Android SDK version on the user's device.
+    android_sdk_version = datastore_services.IntegerProperty(
+        required=False, indexed=True)
+    # The rest of the report info collected on Android. Using the JSON here
```

Comment:
You do not need to justify the choice of JSON in this comment. Ditto below.

Should state (also below) the cases in which this is None. Ditto for other fields.

Reply:
Done